### ISSUE TITLE
New room list: Archived view

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#14038]
TBD

**Comment by nadonomy** [comment]
@turt2live added a proposal for this here:

https://www.figma.com/file/rHnynEgrePUtXVear8rGT0/34-Room-List?node-id=1814%3A0

The intention is to:

- Simplify the room list context menus
- Use the 'archive' interaction for rooms you've left (historical), accessed as a dedicated list as per the room list proposal
- Show a low priority sublist at the bottom if low priority rooms exist
- Expose a toggle to manage this in 'Advanced' in room settings

Please let me know if you see any issues with this, otherwise we'll dress up the comps to be pixel perfect.

**Comment by turt2live** [comment]
@nadonomy looks feasible, thanks for taking a look.

The current behaviour is that if a user sets a low priority room to favourite it leaves low priority and vice versa - presumably this is also kept here (ie: setting a favourite room to low priority removes it from favourites)?

I'll merge https://github.com/vector-im/riot-web/issues/14258 into here and keep the needs-design label on here until we have more dressed up designs.

**Comment by nadonomy** [comment]
> The current behaviour is that if a user sets a low priority room to favourite it leaves low priority and vice versa - presumably this is also kept here (ie: setting a favourite room to low priority removes it from favourites)?

Yeah that sounds good, no reason for a room to try to exist in 2 lists at the same time, it fundamentally breaks the model.

**Comment by turt2live** [comment]
@nadonomy also we'll need to know what a 'selected' state looks like for the favourite button (for rooms in the favourites list): https://github.com/vector-im/riot-web/issues/14269#issuecomment-651806906

**Comment by niquewoodhouse** [comment]
I personally like @t3chguy's [solution](https://github.com/vector-im/riot-web/issues/14269#issuecomment-651806906) @nadonomy.


Although I think a more empathetic way of letting people still low priority a room/do secondary things might be to just have a "more" on the primary list.  This could scale to include a bunch of secondary actions.  

<img width="518" alt="Screenshot 2020-07-01 at 15 28 41" src="https://user-images.githubusercontent.com/1587166/86256149-deceff00-bbaf-11ea-9cae-ed4c54ada411.png">

**Comment by niquewoodhouse** [comment]
I don't have data for the use of low priority so don't know how used it is, so above have assumed it's enough to warrant inclusion in a secondary list.  Maybe archive room is also helpful in that list too.

**Comment by niquewoodhouse** [comment]
The proposed archive view still seems to be appropriate, or am I missing something @turt2live @nadonomy ?


<img width="244" alt="Screenshot 2020-07-01 at 15 50 53" src="https://user-images.githubusercontent.com/1587166/86258577-dd530600-bbb2-11ea-8208-12ac422346de.png">

**Comment by turt2live** [comment]
It depends entirely on what is supposed to be in the view @niquewoodhouse. If it's what was previously known as historical, there won't be badges, rooms will disappear into the void from the main list, and we won't really have much information on the rooms at all. They are also loaded from the server, so a spinner of sorts will be required.

**Comment by turt2live** [comment]
we also probably won't have presence info or public room status.

**Comment by turt2live** [comment]
Low priority moved to https://github.com/vector-im/riot-web/issues/14283

**Comment by chagai95** [comment]
Isn't archive a place you can still read your messages? Maybe Rooms You Left is more appropriate for example?  See WhatsApp and Telegram for reference... @turt2live @nadonomy  what are your thoughts?

**Comment by anoadragon453** [comment]
I think what I'd personally like to see from this whole flow is if I'm in a room, I can choose to either "Leave" a room or "Leave and Archive". The former would leave and forget the room automatically. The latter would leave the room and add it to my Archive. The driving reason is that I don't really care about most rooms I leave. I want an explicit action for keeping rooms around, rather than it being the default.

This doesn't really address the fact that being "kicked" or "banned" from a room would also add it to your Archive. In those cases, perhaps those rooms don't get added to your Archive. Or (going out of spec now) they get added to a separate place where you can choose whether to forget the room or not (archive what contents you could see).
